restcomm/restcomm.ussd/src/main/java/org/restcomm/connect/ussd/telephony/UssdCallManager.java

```diff
@@ -168,7 +160,13 @@ public void onReceive(final Object message) throws Exception {
             final String method = request.getMethod();
             if ("INVITE".equalsIgnoreCase(method)) {
                 check(request);
-                invite(request);
+                if (request.getContentType().equals("application/vnd.3gpp.ussd+xml")) {
```

Comment:
what is this new condition about? is this related to the original  story?
anyway, this seems a good place to use 415/Unsupported Media Type

Reply:
@jaimecasero this is not new, the content type check was moved to there from another point of the same class.
Agree about the response code